titus-client/src/main/java/com/netflix/titus/runtime/endpoint/v3/grpc/GrpcJobManagementModelConverters.java

```diff
@@ -972,6 +1000,38 @@ public static Capacity toGrpcCapacity(com.netflix.titus.api.jobmanager.model.job
                 .build();
     }
 
+    public static List<com.netflix.titus.grpc.protogen.TaskStatus.ContainerState> toGrpcContainerStateList(List<ContainerState> containerStateList) {
+        List<com.netflix.titus.grpc.protogen.TaskStatus.ContainerState> containerStateListGrpc = new ArrayList<>();
```

Comment:
Check if empty, and return immediately to avoid creating unnecessary objects below?